### PR TITLE
feature: support send authentication msg

### PR DESCRIPTION
**Pull request by slievrly**
Signed-off-by: slievrly <slievrly@163.com>

### Ⅰ. Describe what this PR did

feature: support send authentication msg

### CODE REVIEW

**Review by l81893521 — COMMENTED** [review]
Please solve the conflict.

File: core/src/main/java/io/seata/core/protocol/RegisterTMRequest.java [hunk 1]
```diff
@@ -41,6 +51,20 @@ public RegisterTMRequest() {
      */
     public RegisterTMRequest(String applicationId, String transactionServiceGroup, String extraData) {
         super(applicationId, transactionServiceGroup, extraData);
+        StringBuilder sb = new StringBuilder();
+        if (null != extraData) {
+            sb.append(extraData);
+        }
+        if (transactionServiceGroup != null && !transactionServiceGroup.isEmpty()) {
```
Comment (ph3636) [inline core/src/main/java/io/seata/core/protocol/RegisterTMRequest.java:58]: Do you need to add EXTRA_DATA_SPLIT_CHAR for this block?

File: core/src/main/java/io/seata/core/protocol/RegisterTMRequest.java [hunk 2]
```diff
@@ -41,6 +51,20 @@ public RegisterTMRequest() {
      */
     public RegisterTMRequest(String applicationId, String transactionServiceGroup, String extraData) {
         super(applicationId, transactionServiceGroup, extraData);
```
Comment (ph3636) [inline core/src/main/java/io/seata/core/protocol/RegisterTMRequest.java:53]: Can be placed at the bottom

File: core/src/main/java/io/seata/core/protocol/RegisterTMRequest.java (hunk 1 above)
Comment (ph3636) [reply core/src/main/java/io/seata/core/protocol/RegisterTMRequest.java:58]: 这块是否需要添加EXTRA_DATA_SPLIT_CHAR来进行区分？

File: core/src/main/java/io/seata/core/protocol/RegisterTMRequest.java (hunk 2 above)
Comment (ph3636) [reply core/src/main/java/io/seata/core/protocol/RegisterTMRequest.java:53]: super语句可以放到最下面，super(applicationId, transactionServiceGroup, sb.toString());

**Review by ph3636 — APPROVED** [review]
LGTM

**Review by l81893521 — APPROVED** [review]
LGTM

**Review by wangliang181230 — APPROVED** [review]
LGTM for @ph3636

**Pull request merged**